lib/crawly/manager.ex

```diff
@@ -66,6 +67,25 @@ defmodule Crawly.Manager do
 
     Process.link(request_storage_pid)
 
+    # Add start requests to the requests storage
+    init = spider_name.init(options)
+
+    start_requests_from_req = Keyword.get(init, :start_requests, [])
+
+    start_requests_from_urls =
+      init
+      |> Keyword.get(:start_urls, [])
+      |> Crawly.Utils.requests_from_urls()
+
+    start_requests = start_requests_from_req ++ start_requests_from_urls
+
+    # Split start requests, so it's possible to initialize a part of them in async
+    # manner
+    {start_req, async_start_req} =
+      Enum.split(start_requests, @start_request_split_size)
+
+    :ok = store_requests(spider_name, start_req)
```

Comment:
What are the benefits of adding in the synchronous requests at this stage? Does it solve a race condition for the tests?

IMO it would be clearer to have all request insertions in their own handle_continue block, as before, which makes it easier to locate.  

I do agree with the combining of requests/urls to avoid duplication.

Reply:
I was catching some strange cases with workers (in tests only, in general, I think it's a problem of tests which needs to be rewritten in some cases). So I want to feed the RequestsStorage as soon as possible. But I am getting your point, let me think a bit more about the case.